templates/node.ovc/actions.py

```diff
@@ -89,25 +89,29 @@ def _check_ssh_authorization(service, machine):
         # Authorize ssh key into the machine
         _, vm_info = machine.machineip_get()
         if vm_info['status'] not in ['HALTED', 'PAUSED']:
-            prefab = _ssh_authorize_root(service, vm_info)
+            prefab = _ssh_authorize_root(service, machine, vm_info)
             return prefab
     return False
 
 
-def _ssh_authorize_root(service, vm_info):
+def _ssh_authorize_root(service, machine, vm_info):
     if 'sshkey' not in service.producers:
         raise j.exceptions.AYSNotFound("No sshkey service consumed. please consume an sshkey service")
     service.logger.info("Authorizing ssh key to machine {}".format(vm_info['name']))
-
     sshkey = service.producers['sshkey'][0]
     #make sure that SSH key is loaded
     key_path = j.sal.fs.joinPaths(sshkey.path, sshkey.name)
-    j.clients.ssh.load_ssh_key(key_path, True)
 
     executor = j.tools.executor.getSSHBased(addr=service.model.data.ipPublic, port=service.model.data.sshPort,
                                           timeout=5, usecache=False)
-
-    executor.prefab.system.ssh.authorize("root", sshkey.model.data.keyPub)
+    machineip, machinedict = machine.machineip_get()
+    publicip = machine.space.model['publicipaddress']
+    login = machinedict['accounts'][0]['login']
+    password = machinedict['accounts'][0]['password']
+    sshport = machine.portforwards[0]['publicPort']
+    sshclient = j.clients.ssh.get(
+        addr=publicip, port=sshport, login=login, passwd=password, look_for_keys=False, timeout=300)
+    sshclient.SSHAuthorizeKey(sshkey_name=sshkey.name, sshkey_path=key_path)
     service.model.data.sshAuthorized = True
     service.saveAll()
     return executor.prefab
```

Comment:
why are you creating an executor and returning the prefab of the executor when you already have the machine that has a prefab property ? what is the point of creating this executor now ?

Reply:
the prefab property of the machine is using an executor that will try to connect with user "root" and will expect that an sshkey is already loaded

Reply:
but that is okay bc the code just authorized the root user.